test/test_client_db_hooks.js

```diff
@@ -0,0 +1,70 @@
+import app           from './server';
+import db            from '../src/models/db';
+import {
+  datastreams,
+  DatastreamsEntity,
+  entities,
+  featuresOfInterest,
+  FeaturesOfInterestEntity,
+  historicalLocations,
+  HistoricalLocationsEntity,
+  locations,
+  LocationsEntity,
+  observations,
+  ObservationsEntity,
+  observedProperties,
+  ObservedPropertiesEntity,
+  sensors,
+  SensorsEntity,
+  things,
+  ThingsEntity
+} from './constants';
+import supertest     from 'supertest';
+
+const server           = supertest.agent(app.listen(8890));
+
+db().then(models => {
+  describe('db hooks', () => {
+    beforeEach(done => {
+      models.sequelize.transaction(transaction => {
+        return Promise.all(Object.keys(entities).map(name => {
+          return models[name].destroy({ transaction, where: {} });
+        }));
+      }).then(() => done());
+    });
+
+    [
+     [ datastreams, DatastreamsEntity ], 
+     [ featuresOfInterest, FeaturesOfInterestEntity ],
+     [ historicalLocations, HistoricalLocationsEntity ],
+     [ locations, LocationsEntity ],
+     [ observations, ObservationsEntity ],
+     [ observedProperties, ObservedPropertiesEntity ],
+     [ sensors, SensorsEntity ],
+     [ things, ThingsEntity ]
+    ].forEach(testObj => {
+      it(testObj[0] + ' db \"create\" hooks', done => {
+        models[testObj[0]].create(testObj[1]).then(instance => {
```

Comment:
If this approach is ok, I would add "destroy" and "update" tests to exercise the other hooks (I'm not sure what action will trigger the "Restore" hooks).